mlflow/genai/git_versioning/git_info.py

```diff
@@ -0,0 +1,58 @@
+from dataclasses import dataclass
+
+from typing_extensions import Self
+
+from mlflow.utils.mlflow_tags import (
+    MLFLOW_GIT_BRANCH,
+    MLFLOW_GIT_COMMIT,
+    MLFLOW_GIT_DIRTY,
+)
+
+
+class GitOperationError(Exception):
+    """Raised when a git operation fails"""
+
+
+@dataclass(kw_only=True)
+class GitInfo:
```

Comment:
For databricks, we need to adjust this class. `GitPython` doesn't work without `.git`.